multimodal-search-pdf/pods/PDFcrafter.py

```diff
@@ -0,0 +1,183 @@
+import io
+from typing import Dict, List
+
+import numpy as np
+from jina.executors.segmenters import BaseSegmenter
+
+
+class PDFSegmenter(BaseSegmenter):
+    """
+    :class:`PDFExtractorSegmenter` Extracts data (text and images) from PDF files.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def segment(self, uri: str, buffer: bytes, *args, **kwargs) -> List[Dict]:
+        """
+        Segements PDF files. Extracts data from them.
+
+        Checks if the input is a string of the filename,
+        or if it's the file in bytes.
+        It will then extract the data from the file, creating a list for images,
+        and text.
+
+        :param uri: File name of PDF
+        :type uri: str
+        :param buffer: PDF file in bytes
+        :type buffer: bytes
+        :returns: A list of documents with the extracted data
+        :rtype: List[Dict]
+        """
+
+
+        import fitz
+        import PyPDF2
+
+        if uri:
+            pdf_img = fitz.open(uri)
+            pdf_text = open(uri, 'rb')
```

Comment:
Open all files using the `with` context for better error handling

Reply:
Thanks! I will check this!